data_structures/queue_using_one_stack.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+  @author [Rajat Garg](https://github.com/Rajat379)
+  @file
+
+  Implementation of a Queue using one Stack.
```

Comment:
Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..